RA-20-Create-ViewModel

DenisGubernatorov opened the pull request:
1) Создание структурных слоев
2) Распределение файлов по пакетам, согласно структуре
3) Создан прототип для VIewModel

DenisGubernatorov committed:
RA-20-Create-ViewModel
1)Поправлен радиус кнопок навигации

DenisGubernatorov committed:
RA-20-Create-ViewModel
1)Распределение классов по пакетам.
2) Создан файл  recipeViewModel.kt с прототипом ViewModel экрана и data классом описывающим состояние viewMode

aGroupoid commented on app/src/main/java/com/example/recipesapp/data/Recipe.kt:
```diff
@@ -1,4 +1,4 @@
-package com.example.recipesapp
+package com.example.recipesapp.data
```
Надо было разделить на три слоя. Model это твои бизнес-сущности. В Model слой мы выносим всё, что можно использовать в любых проектах, вне зависимости от их реализации. Если бы мы делали список рецептов в виде чат-бота в тг, то мы бы их перенесли без изменений в новый проект.  Data слой именно про получаение данных из источников

aGroupoid commented on app/src/main/java/com/example/recipesapp/ui/recipes/recipe/recipeViewModel.kt:
```diff
@@ -0,0 +1,18 @@
+package com.example.recipesapp.ui.recipes.recipe
```
Название файла с большой буквы

aGroupoid commented on app/src/main/java/com/example/recipesapp/ui/recipes/recipe/recipeViewModel.kt:
```diff
@@ -0,0 +1,18 @@
+    val rTitle: String? = null,
+    val rIngredients: List<Ingredient> = emptyList(),
+    val rMethod: List<String> = emptyList(),
+    val rImageUrl: String? = null
```
Стейт - состояние твоего динамического ui. Всё что ты описал, действительно меняется, но для этого мы создавали отдельный класс Recipe. Почему просто не сделать один пункт с рецептом. 
По мимо рецептов на экране меняется состояние кнопки, можем менять через ползунок количество порций, надо добавить

DenisGubernatorov committed:
RA-20-Create-ViewModel
1) Изменение свойств для RecipeViewModelState. Свойства, относящиеся к рецепту, получаются из объекта Recipe. Добавлено сохранение isFavorite и количества порций
2) Перенос классов относящихся к модели данных в слой модели

aGroupoid reviewed (approved):
Молодец, всё верно!
